### ISSUE TITLE
Feature request: cost based shopping + driving (country)range + price (add-on?) 

### ISSUE THREAD

**Issue by mr4hughz** [grocy/grocy#1151]
In europe we have a lot of specific products that are only available in certain countries or are different sized packaging per country. Also a product can vary per store as well . also "recycling/return fee"  10cents per bottle/crate to return to store 

it would be nice if it would be possible to show savings or extra spending .. or money that can be collected by returning based on input .. think as a "what if i bought product x in another story / country is that worth the trip ? 

I think comparing prices might be a nice addition because pricing wars are not always that transparant. 

Yes in some cases the product quality or personal prefference differs that's an other feature request..

**Comment by berrnd** [comment]
Why not try it to keep it simple:
Use different stores and include the "driving price" in the tracked price.
Price comparision is currently possible via the diagram on the product card, more analysis things are tracked in #64 (my personal biggest feature request for grocy since ever).

About the "recycling-return-fee", of course we could add to track that.
Would also add another thing you would need to track - don't know if it's worth the benefit, added complexity, and so on...

**Comment by berrnd** [comment]
The "recycling-return-fee" would be possible genericly by using Userfields, once #443 is done (and maybe #413).

**Comment by berrnd** [comment]
Cross-referencing https://github.com/grocy/grocy/issues/443#issuecomment-1083298204 here - stock entries can now have notes and Userfields:

> ### New feature: Notes and Userfields for stock entries
> 
> - Stock entries can now have notes
>   - For example to distinguish between same, yet different products (e.g. having only a generic product "Chocolate" and note in that field what special one it is exactly this time)
>   - => New field on the purchase and inventory page
>   - => New column on the stock entries and stock journal page
>   - => Visible also in the "Use a specific stock item" dropdown on the consume and transfer page
> - Additionally it's also possible to add arbitrary own fields by using Userfields
>   - => Configure the desired Userfields for the entity `stock`
>   - => Those Userfields are then visible on the same places as mentioned above for the built-in "Note" field
